src/shared/services/auth/services/keycloak.ts

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2020 - 2021 Gemeente Amsterdam
+import keycloakJS, { KeycloakInitOptions, KeycloakInstance } from 'keycloak-js'
+import configuration from 'shared/services/configuration/configuration'
+
+const AUTH_REDIRECT_URI = `${global.location.protocol}//${global.location.host}/manage/incidents`
+const SILENT_CHECK_SSO_REDIRECT_URI = `${window.location.origin}/assets/html/silent-check-sso.html`
+const OAUTH_DOMAIN_KEY = 'oauthDomain' // Domain that is used for login
+
+const storage = global.localStorage || global.sessionStorage
+
+class Keycloak {
+  private keycloak: KeycloakInstance
+  private refreshIntervalId: number | null = null
+
+  constructor() {
+    this.keycloak = keycloakJS({
```

Comment:
Shouldn't `this.keycloak` be a singleton?

Reply:
By making it a singleton, this class becomes really tricky to unit test. I prefer keeping the current implementation in place and only use the `Keycloak` class in the `auth.ts` file, which is our go-to file for authentication in the app.